webapp/src/main/java/com/box/l10n/mojito/service/tm/search/TextUnitSearcher.java

```diff
@@ -284,6 +287,14 @@ NativeCriteria getCriteriaForSearch(TextUnitSearcherParameters searchParameters)
             conjunction.add(new NativeEqExpFix("atu.branch_id", searchParameters.getBranchId()));
         }
 
+        if (!Strings.isNullOrEmpty(searchParameters.getSkipTextUnitWithPattern())) {
```

Comment:
why special case for the empty string? you may want every text unit that have a non empty name. It the param is present/not null then just add the filter?

Reply:
Updated